code/game/gamemodes/wizard/soulstone.dm

```diff
@@ -200,17 +205,36 @@
 		..()
 
 /obj/item/soulstone/attack_self(mob/living/user)
-	if(!in_range(src, user))
+	var/mob/living/simple_animal/shade/S = locate(/mob/living/simple_animal/shade) in contents
+	if(!in_range(src, user) || !S)
 		return
 
-	if(!can_use(user))
-		user.Weaken(10 SECONDS)
-		user.emote("scream")
-		to_chat(user, "<span class='userdanger'>Your body is wracked with debilitating pain!</span>")
+	if(can_use(user))
+		release_shades(user)
 		return
 
-	release_shades(user)
-	return
+	if(!user.mind.isholy)
+		to_chat(user, "<span class='notice'>The shard feels too tough to shatter, you are not holy enough to free its captive!</span>")
+		return
+
+	if(do_after_once(user, 5 SECONDS, FALSE, src))
+		user.visible_message("[user] shatters the soulstone apart! Releasing [src] from their prison!", "You shatter the soulstone holding [src], binding them free!", "You hear something shatter with a ghastly crack.")
```

Comment:
Not sure about the messaging here. I think src would probably refer to the soulstone, making this somewhat misleading.